packages/headless/src/controllers/search-box/headless-search-box.ts

```diff
@@ -205,11 +205,16 @@ export function buildSearchBox(
 
     selectSuggestion(value: string) {
       dispatch(selectQuerySuggestion({id, expression: value}));
-      performSearch(logQuerySuggestionClick({id, suggestion: value}));
+      performSearch(logQuerySuggestionClick({id, suggestion: value})).then(
+        () => {
+          dispatch(clearQuerySuggest({id}));
+        }
```

Comment:
If query suggestions are supposed to be cleared after every search, a simplification might be to add a reducer case on `executeSearch.fulfilled`. I _think_ the analytics call would be dispatched before the state changes.

It would be similar to how facet search results are cleared on every search. It's a behaviour change though, so it could be considered breaking from the perspective of someone dispatching actions directly.

Another idea would be to adjust the log analytics functions to accept all their metadata upfront (i.e. make them pure functions). This would remove order issues, which [we've seen in the past](https://coveord.atlassian.net/browse/KIT-412). It would make the functions harder to use if someone is dispatching them themselves, but as you said, there's a chance of getting the order wrong as it is currently. Unless there are other options, I feel this path would be best long-term,

We can push the current fix, but I think we should address the deeper issue soon,

Reply:
> If query suggestions are supposed to be cleared after every search, a simplification might be to add a reducer case on executeSearch.fulfilled. I think the analytics call would be dispatched before the state changes.

No, unfortunately, (as explained in my original comment in this PR), it was part of the problem before.

> Another idea would be to adjust the log analytics functions to accept all their metadata upfront (i.e. make them pure functions).

Yup, exactly. This is what I proposed as an alternative in my original comment.